### PR TITLE
[INLONG-10635][Manager] Optimize the installer configuration process

### LINKED ISSUE #10635: [Improve][Manager] Optimize the installer configuration process
### Description

Optimize the installer configuration process.

### InLong Component

InLong Manager

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by fuweng11**
Fixes #10635

### Motivation

Optimize the installer configuration process.

### Modifications

Optimize the installer configuration process.
1.When setting up the agent node, module config will be loaded into agent_task_config.
2.If you actively install the installer, it will obtain the default module configuration.

### CODE REVIEW

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cluster/node/AgentClusterNodeOperator.java [hunk 1]
```diff
@@ -88,4 +164,94 @@ protected void setTargetEntity(ClusterNodeRequest request, InlongClusterNodeEnti
...
+            LOGGER.info("begin to update module config for cluster name={}, ip={}", clusterName, ip);
+            if (StringUtils.isBlank(clusterName) || StringUtils.isBlank(ip)) {
+                LOGGER.info("no need to update module config when ip or cluster name is null");
+                return;
+            }
+            configRequest.setLocalIp(ip);
+            configRequest.setClusterName(clusterName);
+            ConfigResult configResult = loadModuleConfigs(configRequest);
+            AgentTaskConfigEntity existEntity = agentTaskConfigEntityMapper.selectByIdentifier(ip, clusterName);
+            AgentTaskConfigEntity agentTaskConfigEntity = new AgentTaskConfigEntity();
```
Comment (vernedeng) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cluster/node/AgentClusterNodeOperator.java:180]:
```suggestion
            AgentTaskConfigEntity agentTaskConfigEntity = existEntity == null ? new AgentTaskConfigEntity() : CommonBeanUtils.copyProperties(existEntity, AgentTaskConfigEntity::new, true);
```

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cluster/node/AgentClusterNodeOperator.java [hunk 2]
```diff
@@ -88,4 +164,94 @@ protected void setTargetEntity(ClusterNodeRequest request, InlongClusterNodeEnti
...
+
+    private List<ModuleConfig> getModuleConfigs(AgentClusterNodeDTO dto) {
+        List<Integer> moduleIdList = dto.getModuleIdList();
+        List<ModuleConfig> configs = new ArrayList<>();
+        if (CollectionUtils.isEmpty(moduleIdList)) {
+            return configs;
+        }
+        for (Integer moduleId : moduleIdList) {
+            ModuleConfigEntity moduleConfigEntity = moduleConfigEntityMapper.selectByPrimaryKey(moduleId);
+            ModuleConfig moduleConfig = CommonBeanUtils.copyProperties(moduleConfigEntity, ModuleConfig::new);
```
Comment (vernedeng) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cluster/node/AgentClusterNodeOperator.java:232]: may produce NPE?

Comment (fuweng11) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cluster/node/AgentClusterNodeOperator.java:232]: Fixed.

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cluster/node/AgentClusterNodeOperator.java (hunk 1 above)
Comment (fuweng11) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cluster/node/AgentClusterNodeOperator.java]: Fixed.

**Review by vernedeng — DISMISSED** [review]

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cluster/node/AgentClusterNodeOperator.java [hunk 3]
```diff
@@ -88,4 +163,95 @@ protected void setTargetEntity(ClusterNodeRequest request, InlongClusterNodeEnti
...
     }
+
+    public void updateModuleConfig(String ip, String clusterName) {
+        try {
+            ConfigRequest configRequest = new ConfigRequest();
+            LOGGER.info("begin to update module config for cluster name={}, ip={}", clusterName, ip);
+            if (StringUtils.isBlank(clusterName) || StringUtils.isBlank(ip)) {
+                LOGGER.info("no need to update module config when ip or cluster name is null");
+                return;
+            }
```
Comment (aloyszhang) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cluster/node/AgentClusterNodeOperator.java:174]: blank check move to the top of the method

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AgentServiceImpl.java [hunk 4]
```diff
@@ -323,6 +300,15 @@ public void reload() {
                         agentConfigInfo.setVersion(agentTaskConfigEntity.getVersion());
                         newAgentConfigMap.putIfAbsent(key, agentConfigInfo);
                     }
+                    ConfigResult configResult =
+                            JsonUtils.parseObject(agentTaskConfigEntity.getModuleParams(), ConfigResult.class);
+                    if (configResult != null) {
+                        configResult.setVersion(agentTaskConfigEntity.getVersion());
+                        newInstallerConfigMap.putIfAbsent(key, configResult);
+                        if (Objects.equals(agentTaskConfigEntity.getAgentIp(), "9.135.95.77")) {
+                            LOGGER.info("Test get md5={}, configresult={}", configResult.getMd5(), configResult);
```
Comment (aloyszhang) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AgentServiceImpl.java:309]:
```suggestion
                            LOGGER.info("Test get md5={}, configResult={}", configResult.getMd5(), configResult);
```

Comment (fuweng11) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/AgentServiceImpl.java]: Fixed.

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cluster/node/AgentClusterNodeOperator.java (hunk 3 above)
Comment (fuweng11) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cluster/node/AgentClusterNodeOperator.java:172]: Fixed.

**Review by aloyszhang — DISMISSED** [review]

**Review by justinwwhuang — APPROVED** [review]
LGTM

**Review by justinwwhuang — APPROVED** [review]

**Review by vernedeng — APPROVED** [review]

**Review by aloyszhang — APPROVED** [review]
LGTM

**Pull request merged**